sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -567,13 +575,7 @@ private RntbdRequestRecord addPendingRequestRecord(final ChannelHandlerContext c
             final Timeout pendingRequestTimeout = record.newTimeout(timeout -> {
 
                 // We don't wish to complete on the timeout thread, but rather on a thread doled out by our executor
-                final EventExecutor executor = context.executor();
-
-                if (executor.inEventLoop()) {
-                    record.expire();
-                } else {
-                    executor.next().execute(record::expire);
-                }
+                requestExpirator.execute(record::expire);
```

Comment:
Clarification(non-blocking): Any idea if netty duplex channel support timeout OOB?

Reply:
Can be discussed offline.